### PR TITLE
Add unit test for multipath.CheckDevice.

### PR DESCRIPTION
**Pull request by LianWaiYuChanChan**
As title.

### CODE REVIEW

**Comment by LianWaiYuChanChan** [comment]
@peter-wangxu  OK. BTW, can we  mock the command return status?

**Comment by peter-wangxu** [comment]
@LianWaiYuChanChan I just implemented the mocked output support for now.
You could help add the support for mocked error later or in this patch

**Comment by LianWaiYuChanChan** [comment]
@peter-wangxu  Got it. I am going to add this support in later patch.

File: model/model.go [hunk 1]
```diff
@@ -257,7 +257,10 @@ func (s *HBA) Parse() []HBA {
 func (s *HBA) getOutput() string {
 	cmd := s.GetCommand()
 	out, err := executor.Command(cmd[0], cmd[1:]...).CombinedOutput()
+	fmt.Println(err)
```
Comment (peter-wangxu) [inline model/model.go]: remove debug log here

File: model/model.go [hunk 2]
```diff
@@ -257,7 +257,10 @@ func (s *HBA) Parse() []HBA {
...
+	fmt.Println(out)
 	if nil != err {
+		fmt.Println("nil != err")
```
Comment (peter-wangxu) [inline model/model.go]: Ditto

File: test/mock_util.go [hunk 3]
```diff
@@ -66,17 +70,44 @@ func (m *MockCmd) mockOutput() ([]byte, error) {
...
 	fileName = strings.Replace(fileName, "\\", "_", -1)
 	fileName = fmt.Sprintf("%s%s.txt", getMockDir(), fileName)
-	file, err := os.Open(fileName)
-	if nil != err {
+
+	// open a file
+	if file, err := os.Open(fileName); err == nil {
+
+		// make sure it gets closed
+		defer file.Close()
```
Comment (peter-wangxu) [inline test/mock_util.go]: nice!

File: test/mock_util.go [hunk 4]
```diff
@@ -66,17 +70,44 @@ func (m *MockCmd) mockOutput() ([]byte, error) {
...
+		scanner := bufio.NewScanner(file)
+		var buffer bytes.Buffer
+		cmdStatus := "-1"
+		isFirstLine := true
+		for scanner.Scan() {
+			if isFirstLine {
+				cmdStatus = scanner.Text()
+				isFirstLine = false
+			} else {
+				buffer.WriteString(scanner.Text() + "\n")
```
Comment (peter-wangxu) [inline test/mock_util.go]: why the `\n` is needed, the original text contains the end of line.

File: model/model.go (hunk 2 above)
Comment (LianWaiYuChanChan) [inline model/model.go]: OK.

File: model/model.go (hunk 1 above)
Comment (LianWaiYuChanChan) [inline model/model.go]: OK.

File: test/mock_util.go (hunk 4 above)
Comment (LianWaiYuChanChan) [inline test/mock_util.go]: It seems that scanner.Text() won't contain "\n".

**Pull request merged**